lib/codegen/meta/src/isa/mod.rs

```diff
@@ -0,0 +1,399 @@
+use cranelift_entity::EntityRef;
+mod x86;
+mod arm32;
+//mod arm64;
+//mod riscv;
+
+fn is_power_of_two(n: u8) -> bool {
+    return n > 0 && n & (n - 1) == 0;
+}
+
+fn next_power_of_two(x: u8) -> u8 {
+    1 << (8 - x.leading_zeros())
+}
+
+/// Represents known ISA target.
+#[derive(Copy, Clone)]
+pub enum Isa {
+    Riscv,
+    X86,
+    Arm32,
+    Arm64,
+}
+
+impl Isa {
+    /// Creates isa target using name.
+    pub fn new(name: &str) -> Option<Self> {
+        Isa::all()
+            .iter()
+            .cloned()
+            .filter(|isa| isa.name() == name)
+            .next()
+    }
+
+    /// Creates isa target from arch.
+    pub fn from_arch(arch: &str) -> Option<Isa> {
+        Isa::all()
+            .iter()
+            .cloned()
+            .filter(|isa| isa.is_arch_applicable(arch))
+            .next()
+    }
+
+    /// Returns all supported isa targets.
+    pub fn all() -> [Isa; 4] {
+        [Isa::Riscv, Isa::X86, Isa::Arm32, Isa::Arm64]
+    }
+
+    /// Returns name of the isa target.
+    pub fn name(&self) -> &'static str {
+        match *self {
+            Isa::Riscv => "riscv",
+            Isa::X86 => "x86",
+            Isa::Arm32 => "arm32",
+            Isa::Arm64 => "arm64",
+        }
+    }
+
+    /// Checks if arch is applicable for the isa target.
+    fn is_arch_applicable(&self, arch: &str) -> bool {
+        match *self {
+            Isa::Riscv => arch == "riscv",
+            Isa::X86 => ["x86_64", "i386", "i586", "i686"].contains(&arch),
+            Isa::Arm32 => arch.starts_with("arm") || arch.starts_with("thumb"),
+            Isa::Arm64 => arch == "aarch64",
+        }
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct RegBankIndex(u32);
+entity_impl!(RegBankIndex);
+
+pub struct RegBank {
+    pub name: &'static str,
+    doc: &'static str,
+    pub reg_prefix: &'static str,
+    pub reg_names: Option<Vec<&'static str>>,
+    pub first_unit: u8,
+    pub units: u8,
+    pub pressure_tracking: bool,
+    classes: Vec<RegClassIndex>,
+    pub toprcs: Vec<RegClassIndex>,
+}
+
+impl RegBank {
+    fn new(
+        name: &'static str,
+        doc: &'static str,
+        reg_prefix: &'static str,
+        reg_names: Option<Vec<&'static str>>,
+        first_unit: u8,
+        units: u8,
+        pressure_tracking: bool,
+    ) -> Self {
+        RegBank {
+            name,
+            doc,
+            reg_prefix,
+            reg_names,
+            first_unit,
+            units,
+            pressure_tracking,
+            classes: Vec::new(),
+            toprcs: Vec::new(),
+        }
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct RegClassIndex(u32);
+entity_impl!(RegClassIndex);
+
+pub struct RegClass {
+    pub name: &'static str,
+    pub width: u8,
+    bitmask: u128,
+    pub bank: RegBankIndex,
+    subclasses: Vec<RegClassIndex>,
+    pub toprc: RegClassIndex,
+    pub index: RegClassIndex,
+}
```

Comment:
Could you sort these fields to more closely follow the order of fields in the `lib/codegen/src/isa/registers.rs` version of `RegClass`, so that it's easier to compare the two?